### PR TITLE
Create files for Firewall Policy Rule - override

### PR DESCRIPTION
**Pull request by diogoEsteves42**
Solves https://github.com/hashicorp/terraform-provider-google/issues/14589

Created one scenario of tests with address_groups for beta and another without the address_groups for GA.

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
```

### CODE REVIEW

**Comment by diogoEsteves42** [comment]
Hi @shuyama1 I hope you are going well.
Could you take a look at this PR, please?

**Review by shuyama1 — COMMENTED** [review]
Thanks for making the change. I made a first pass and just want to make sure if I understand the purpose of this PR - it's only adding new tests and correcting documentation but not adding any new features? I changed the release note based on this and just want to make sure I don't miss anything.

File: tpgtools/overrides/compute/samples/firewallpolicyrule/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# meta.yaml
```
[the remark this answers is not in the document]
Comment (diogoEsteves42) [reply tpgtools/overrides/compute/samples/firewallpolicyrule/meta.yaml:1]: Hi @shuyama1, thanks for the review. This file `mmv1/third_party/terraform/website/docs/r/compute_firewall_policy_rule.html.markdown` I thought it was automatically generated by the tpgtools, because there is a header mentioning that it's "Auto Generated Code".

**Comment by diogoEsteves42** [comment]
@shuyama1, correct, just adding new tests and updating the doc for the resource, not adding any new features.

File: tpgtools/overrides/compute/beta/firewall_policy_rule.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-- type: DIFF_SUPPRESS_FUNC
-  field: target_resources
+- type: CUSTOM_RESOURCE_NAME
   details:
-    diffsuppressfunc: compareSelfLinkOrResourceName
```
[the remark this answers is not in the document]
Comment (diogoEsteves42) [reply tpgtools/overrides/compute/beta/firewall_policy_rule.yaml]: At the first time I was just following the example from the NetworkFirewall resource, but returned the code to the original.

File: tpgtools/overrides/compute/beta/firewall_policy_rule.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-- type: DIFF_SUPPRESS_FUNC
-  field: target_resources
+- type: CUSTOM_RESOURCE_NAME
```
[the remark this answers is not in the document]
Comment (diogoEsteves42) [reply tpgtools/overrides/compute/beta/firewall_policy_rule.yaml]: same.

File: tpgtools/overrides/compute/firewall_policy_rule.yaml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-- type: DIFF_SUPPRESS_FUNC
```
[the remark this answers is not in the document]
Comment (diogoEsteves42) [reply tpgtools/overrides/compute/firewall_policy_rule.yaml:1]: same.

File: tpgtools/overrides/compute/samples/firewallpolicyrule/meta.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (diogoEsteves42) [reply tpgtools/overrides/compute/samples/firewallpolicyrule/meta.yaml:1]: @shuyama1 this doc was updated few days ago with the PR mentioned, no this PR just add tests for the scenarios and there is updates in small parts of the doc.

**Comment by diogoEsteves42** [comment]
@shuyama1 the PR that I've mentioned it updates the documentation part, this PR now adds some minor updates in the doc and add the tests scenarios.

**Comment by shuyama1** [comment]
@diogoEsteves42 Thanks for the update. I'll start reviewing the PR

File: tpgtools/overrides/compute/beta/firewall_policy_rule.yaml [hunk 5]
```diff
@@ -1,4 +1,3 @@
 - type: DIFF_SUPPRESS_FUNC
-  field: target_resources
```
Comment (shuyama1) [inline tpgtools/overrides/compute/beta/firewall_policy_rule.yaml:2]: Removing this will result in the diff suppress funtion `CompareSelfLinkOrResourceName` being removed from the field `target_resources` for providers (https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7940-old..auto-pr-7940). Is this removal intentional?

**Review by shuyama1 — COMMENTED** [review]
The test looks good! But it looks like now the web doc is generated through DCL engine instead of the handwritten file that modified in https://github.com/GoogleCloudPlatform/magic-modules/pull/8025 

https://github.com/modular-magician/terraform-provider-google-beta/blob/9a9d71abb78e1c7d6434e3a4c94b95a4fe4a0a98/website/docs/r/compute_firewall_policy_rule.html.markdown is what the downstream web documentation is going to looks like. Please confirm if this is what we want.

File: tpgtools/overrides/compute/beta/firewall_policy_rule.yaml (hunk 5 above)
Comment (diogoEsteves42) [reply tpgtools/overrides/compute/beta/firewall_policy_rule.yaml:2]: Yes, I've removed that, I'm using this file as example: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/tpgtools/overrides/compute/network_firewall_policy_rule.yaml
And they don't use this field there. But it makes more sense to keep that in the code. Will fix that.

**Comment by diogoEsteves42** [comment]
@shuyama1 the doc should look like that, for sure. I see that DCL is not generating everything from zero, it's taking the file that exists and adding to them some minor changes from DCL (if they update the DCL resource these changes will be updated in the doc without opening another PR here), I see that looking to the diffs in the file
https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7940-old..auto-pr-7940

Does it make sense?

**Review by shuyama1 — APPROVED** [review]
Thank you!
